Lib/unittest/mock.py

```diff
@@ -2754,7 +2754,19 @@ def create_autospec(spec, spec_set=False, instance=False, _parent=None,
         raise InvalidSpecError(f'Cannot autospec a Mock object. '
                                f'[object={spec!r}]')
     is_async_func = _is_async_func(spec)
-    _kwargs = {'spec': spec}
+
+    placeholder = object()
+    entries = [(entry, placeholder) for entry in dir(spec)]
+    # Not using `is_dataclass` to avoid an import of dataclasses module
+    # for types that don't need that.
+    if is_type and instance and hasattr(spec, '__dataclass_fields__'):
+        from dataclasses import fields
```

Comment:
What leads to the local import here? Why are you looking to avoid import of the dataclasses module?

Reply:
Still curious about the answer to this question, worried I'm missing something?

Reply:
This was changed to a regular top level import (for some modules, minimising eager imports is important for startup performance, but mock isn't one of them).

Reply:
Ah right, yeah, I wish we could find a better story on that. Local imports are ugly, but I remember the deferred import peps running into problems :-/